src/resources/cpp/lambdas.md

```diff
@@ -0,0 +1,365 @@
+
+static_assert(not std::same_as<decltype(foo), decltype(bar)>);
+```
+
+&nbsp;
+
+### Capture list
+
+All lambda expressions begin with a capture list. The capture list specifies which variables to capture from the
+surrounding scope:
+
+```cpp
+int x = 10;
+
+auto lambda = [x](int y) { return x + y; };
+
+lambda(5) // 15
+```
+
+- `[]` - Empty capture list.
+- `[=]` - Automatically capture variables that are used in the lambda body by value.
+  - Mutually exclusive with `[&]`.
+  - Does not implicitly capture `this` if used in a class.
+- `[&]` - Automatically capture variables that are used in the lambda body by reference.
+  - Mutually exclusive with `[=]`.
+- `[x]` - Capture only `x` by value.
+- `[&x]` - Capture only `x` by reference.
+- `[x...]` - Capture a pack `x` by value.
+- `[&x...]` - Capture a pack `x` by reference.
+- `[this]` - Capture `*this` by reference.
+- `[x = y]` - Define a local variable `x` and initialize it to `y`.
+
+Different captures can be mixed together:
+
+```cpp
+int a = 1;
+int b = 2;
+int c = 3;
+
+auto foo = [=, &b, d = c]() {
+	// `a` is implicitly captured by value,
+	// `b` is explicitly captured by reference, and
+	// `d` is initialized to `c`.
+	return a + b + d;
+};
+```
+
+However, `[=]` may only be followed by captures by reference and `[&]` may only be followed by captures by value.
+
+Implicit captures can be nested multiple times:
+
+```cpp
+int x = 0;
+
+[&]() {
+	[&]() {
+		x = 5;
+	}();
+}();
+```
+
+&nbsp;
+
+### Parameters
+
+Lambda parameters work the same way as normal function parameters, and `auto` parameters make a lambda's `operator()`
+implicitly templated.
+
+If a lambda takes no parameters, the parameter list may be omitted entirely:
+
+```cpp
+auto very_important_number = [] { return 4; };
+```
+
+Finally, the explicit `this` parameter can also be in lambdas since C++23:
+
+```cpp
+auto fibonacci = [](this auto self, int n) {
+	if (n < 2) return n;
+	return self(n - 1) + self(n - 2);
+};
+```
+
+&nbsp;
+
+### Return type
+
+A lambda's return type may be specified with an arrow:
+
+```cpp
+auto add = [](int x, int y) -> int { return x + y; };
+```
+
+Omitting the return type is the same as writing `-> auto`.
+
+&nbsp;
+
+### Specifiers
+
+- `constexpr` - Explicitly specifies that a lambda's `operator()` is a
+  [constexpr function](https://en.cppreference.com/w/cpp/language/constexpr#constexpr_function).
+  - Mutually exclusive with `consteval`.
+  - Lambdas are implicitly marked `constexpr`, if possible.
+- `consteval` - Makes a lambda's `operator()` an
+  [immediate function](https://en.cppreference.com/w/cpp/language/consteval).
+  - Mutually exclusive with `constexpr`.
+- `static` - Makes a lambda's `operator()` a
+  [static member function](https://en.cppreference.com/w/cpp/language/static#Static_member_functions).
+  - Mutually exclusive with `mutable`.
+  - Cannot be used if the captures list is not empty, or an explicit `this` parameter is present.
```

Comment:
I think it would be helpful to describe why this might be useful. Also it might be worth noting the lambda -> function pointer trick in some other section, and how this works even without `static`:

```cpp
auto* ptr = +[]{ return 4; };
```